api/src/main/java/ca/bc/gov/educ/api/grad/report/service/SchoolReportsService.java

```diff
@@ -0,0 +1,132 @@
+    if (schoolReport != null && schoolReport.getReport() != null) {
+      byte[] reportByte = Base64.decodeBase64(schoolReport.getReport().getBytes(StandardCharsets.US_ASCII));
+      ByteArrayInputStream bis = new ByteArrayInputStream(reportByte);
+      return ResponseEntity
+              .ok()
+              .contentType(MediaType.APPLICATION_PDF)
+              .body(new InputStreamResource(bis));
+    }
+    return null;
+  }
+
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public ResponseEntity<Void> updateSchoolReports(UUID schoolOfRecordId, String reportTypeCode) {
+    Optional<SchoolReportsEntity> optEntity = schoolReportsRepository.findBySchoolOfRecordIdAndReportTypeCode(schoolOfRecordId, reportTypeCode);
+    if (optEntity.isPresent()) {
+      SchoolReportsEntity ent = optEntity.get();
+      ent.setUpdateDate(null);
+      ent.setUpdateUser(null);
+      schoolReportsRepository.save(ent);
+      return ResponseEntity.noContent().build();
+    }
+    logger.warn("No school report found to update for schoolOfRecordId {} and reportTypeCode {}", schoolOfRecordId, reportTypeCode);
+    return ResponseEntity.notFound().build();
+  }
+
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public SchoolReports saveSchoolReports(SchoolReports schoolReports) {
+    SchoolReportsEntity toBeSaved = schoolReportsTransformer.transformToEntity(schoolReports);
+    Optional<SchoolReportsEntity> existingEntity = schoolReportsRepository.findBySchoolOfRecordIdAndReportTypeCode(schoolReports.getSchoolOfRecordId(), schoolReports.getReportTypeCode());
+    if (existingEntity.isPresent()) {
+      SchoolReportsEntity gradEntity = existingEntity.get();
+      gradEntity.setUpdateDate(null);
+      gradEntity.setUpdateUser(null);
+      if (schoolReports.getReport() != null) {
+        gradEntity.setReport(schoolReports.getReport());
+      }
+      return schoolReportsTransformer.transformToDTO(schoolReportsRepository.save(gradEntity));
+    } else {
+      return schoolReportsTransformer.transformToDTO(schoolReportsRepository.save(toBeSaved));
+    }
+  }
+
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public ResponseEntity<Void> deleteSchoolReport(UUID schoolOfRecordId, String reportTypeCode) {
+    Optional<SchoolReportsEntity> optEntity = schoolReportsRepository.findBySchoolOfRecordIdAndReportTypeCode(schoolOfRecordId, reportTypeCode);
+    if (optEntity.isPresent()) {
+      schoolReportsRepository.delete(optEntity.get());
+      return ResponseEntity.noContent().build();
+    }
+    logger.warn("No school report found to delete for schoolOfRecordId {} and reportTypeCode {}", schoolOfRecordId, reportTypeCode);
+    return ResponseEntity.notFound().build();
+  }
+
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public ResponseEntity<Void> deleteAllSchoolReportsByType(String reportTypeCode) {
```

Comment:
Move response entity up to controller level

Reply:
updated